src/Client.php

```diff
@@ -423,40 +424,55 @@ private function sharedContextFactory($initializer, $argv = null)
     }
 
     /**
-     * Creates a new pipeline context and returns it, or returns the results of
-     * a pipeline executed inside the optionally provided callable object.
+     * Creates a new pipeline context.
      *
-     * @param mixed ... Array of options, a callable for execution, or both.
+     * This method returns the pipeline object unless a callable is provided by
+     * the user. In this case the user-supplied callable receives the pipeline
+     * instance as the only argument and the pipeline will be executed by the
+     * client at the end, returning the responses read back from the server.
+     *
+     * @param mixed ... Array of options, a callable for execution, or both
      *
-     * @return Pipeline|array
+     * @return Pipeline|Traversable|array
      */
     public function pipeline(/* arguments */)
     {
         return $this->sharedContextFactory('createPipeline', func_get_args());
     }
 
     /**
-     * Actual pipeline context initializer method.
+     * Pipeline context initializer method.
      *
-     * @param array $options  Options for the context.
-     * @param mixed $callable Optional callable used to execute the context.
+     * @param array    $options  Optional configuration for the context
+     * @param callable $callable Optional callable used to execute the context
      *
-     * @return Pipeline|array
+     * @return Pipeline|Traversable|array
      */
-    protected function createPipeline(array $options = null, $callable = null)
+    protected function createPipeline(array $options = null, callable $callable = null)
     {
-        if (isset($options['atomic']) && $options['atomic']) {
-            $class = 'Predis\Pipeline\Atomic';
-        } elseif (isset($options['fire-and-forget']) && $options['fire-and-forget']) {
-            $class = 'Predis\Pipeline\FireAndForget';
+        if (isset($options['traversable']) && $options['traversable']) {
+            $traversable = true;
         } else {
-            $class = 'Predis\Pipeline\Pipeline';
+            $traversable = false;
         }
```

Comment:
I hope this isn't a nitpick but I'd recommend we make this terse like so:

```php
$traversable = isset($options['traversable']) && $options['traversable'];

// alternatively

$traversable = false;

if (isset($options['traversable']) && $options['traversable']) {
    $traversable = true;
}
```

Reply:
It can also be something like:

```
$traversable = isset($options['traversable']) && $options['traversable'] ? true : false;
```

Reply:
Nitpicking is fine!

And sometimes I still forget that the null coalesce operator in PHP 7 works fine for lookups on undefined indexes:

```php
$traversable = (bool) ($options['traversable'] ?? false);
```

With psysh:

```
>>> $options = [];
=> []
>>> $traversable = (bool) ($options['traversable'] ?? false);
=> false

>>> $options = ['traversable' => false];
=> [
     "traversable" => false,
   ]
>>> $traversable = (bool) ($options['traversable'] ?? false);
=> false

>>> $options = ['traversable' => true];
=> [
     "traversable" => true,
   ]
>>> $traversable = (bool) ($options['traversable'] ?? false);
=> true
```

Can't be more terse than that :smile:

Reply:
LOL!